### PR TITLE
cleanup proxy

### PR DESCRIPTION
**Pull request by aruniverse**
cleanup proxy
- add support for .env.local to add env vars
- in fwded post req, make sure we fwd params if any
- get reqs dont have a body, so remove that

### CODE REVIEW

File: proxy/package.json [hunk 1]
```diff
@@ -12,20 +12,23 @@
...
-    "openid-client": "^4.2.2",
-    "ts-node-dev": "^1.1.1"
+    "openid-client": "^4.2.2"
   },
   "devDependencies": {
+    "@types/dotenv-flow": "^3.1.0",
     "@types/express": "^4.17.9",
     "@types/jsonwebtoken": "^8.5.0",
     "rimraf": "^2.6.2",
+    "ts-node-dev": "^1.1.1",
```
Comment (calebmshafer) [inline proxy/package.json:31]: What's this one for? Does it do a tsc watch and run the server.

File: proxy/src/AuthClient.ts [hunk 2]
```diff
@@ -38,18 +42,26 @@ export class AuthClient {
     return this._client;
   }
 
-  private async generateAccessToken(): Promise<AccessToken> {
+  private async _generateAccessToken(): Promise<AccessToken> {
     const scope = this._scopes;
-    if (scope.includes("openid") || scope.includes("email") || scope.includes("profile") || scope.includes("organization"))
-      throw new Error("Scopes for an Agent cannot include 'openid email profile organization'");
+    if (
+      scope.includes("openid") ||
```
Comment (calebmshafer) [inline proxy/src/AuthClient.ts:48]: silly prettier rules ;)

File: proxy/src/CustomExpressServer.ts [hunk 3]
```diff
@@ -54,34 +60,49 @@ export class CustomExpressServer {
...
-    if (undefined !== this._itwinUrl)
+  private async getNewUrl(
+    ctx: AuthorizedClientRequestContext
+  ): Promise<string> {
+    if (this._itwinUrl) {
       return this._itwinUrl;
+    }
     const urlClient = new UrlDiscoveryClient();
-    this._itwinUrl = await urlClient.discoverUrl(ctx, "iModelJsOrchestrator.K8S", undefined);
+    this._itwinUrl = await urlClient.discoverUrl(
```
Comment (calebmshafer) [inline proxy/src/CustomExpressServer.ts]: We should probably just take out the use of UrlDiscoveryClient. It's going to be deprecated soon.  The URL for the Visualization service is `https://api.bentley.com/imodeljs` at the moment so lets use that.

File: proxy/package.json (hunk 1 above)
Comment (aruniverse) [reply proxy/package.json:31]: yea, exactly. was previously listed as a direct dep, but should be a dev dep.

File: proxy/src/AuthClient.ts (hunk 2 above)
Comment (aruniverse) [reply proxy/src/AuthClient.ts:48]: yea i know... a lot of the changes are from prettier, i can revert if you want, just lmk

Comment (calebmshafer) [reply proxy/src/AuthClient.ts:48]: Nah that's fine you can leave them, just thought i'd complain a bit :)

File: proxy/.env [hunk 4]
```diff
@@ -6,3 +6,4 @@
 
 CLIENT_ID=
 CLIENT_SECRET=
+VISUALIZATION_URL=
```
Comment (calebmshafer) [inline proxy/.env:9]: Where is this used?

Comment (calebmshafer) [inline proxy/.env]: Can you default this to `https://api.bentley.com/imodeljs`?

Comment (aruniverse) [reply proxy/.env]:
```suggestion
VISUALIZATION_URL=https://api.bentley.com/imodeljs
```

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**